src/level/level.cpp

```diff
@@ -238,22 +247,112 @@ namespace level {
+                if(s->getPosition().colonna < levelManager->getPlayer()->getPosition().colonna){
+                    for(int i = (s->getPosition().colonna + 1); i <  (levelManager->getPlayer()->getPosition().colonna); i++){
+                        collision = getCollision(Position (s->getPosition().riga, i), levelManager);
+                        enums::CollisionType type = enums::CollisionType::NONE;
+                        if (collision != nullptr) type = collision->getCollisionType();
+                        if(type != enums::CollisionType::NONE){
+                            firable = false;
+                        }
+                    }
+                    if(firable){
+                        s->resetShootCoolDown();
+                        logger_.info("enemy firing a bullet");
+                        Position bulletPosition = Position(s->getPosition().riga, s->getPosition().colonna + 1);
+                        weapon::Bullet *bullet = new weapon::Bullet(bulletPosition, enums::Direction::RIGHT, s->getAttack());
+                        this->addBullet(bullet);
+                    }
+                }
+                else{
+                    for(int i = (s->getPosition().colonna - 1); i >  (levelManager->getPlayer()->getPosition().colonna); i--){
+                        collision = getCollision(Position (s->getPosition().riga, i), levelManager);
+                        enums::CollisionType type = enums::CollisionType::NONE;
+                        if (collision != nullptr) type = collision->getCollisionType();
+                        if(type != enums::CollisionType::NONE){
+                            firable = false;
+                        }
+                    }
+                    if(firable){
+                        s->resetShootCoolDown();
+                        logger_.info("enemy firing a bullet");
+                        Position bulletPosition = Position(s->getPosition().riga, s->getPosition().colonna - 1);
+                        weapon::Bullet *bullet = new weapon::Bullet(bulletPosition, enums::Direction::LEFT, s->getAttack());
+                        this->addBullet(bullet);
+                    }
+                }
+            }
+            if(s->getPosition().colonna == levelManager->getPlayer()->getPosition().colonna){
+                if(s->getPosition().riga < levelManager->getPlayer()->getPosition().riga){
+                    for(int i = (s->getPosition().riga + 1); i <  (levelManager->getPlayer()->getPosition().riga); i++){
+                        collision = getCollision(Position (i, s->getPosition().colonna), levelManager);
+                        enums::CollisionType type = enums::CollisionType::NONE;
+                        if (collision != nullptr) type = collision->getCollisionType();
+                        if(type != enums::CollisionType::NONE){
+                            firable = false;
+                        }
+                    }
```

Comment:
Questo è uguale a sopra, c'è moltissimo codice ripetuto in questa parte, vorrei cercare di renderlo più chiaro nell'intento.
Qui stai facendo una cosa speculare alla precedente. 

Per evitare di ripetere questo potresti assegnare a una variabile il minimo, all'altra il massimo e ciclare in questo modo.

Provo a scriverti uno pseudocodice in modo che possa essere chiaro.

```cpp
int minColonna = min(posizione-colonna, posizione-giocatore)
int maxcolonna = max(posizione-coonna, posizione-giocatore)
[poi fai di nuovo il check in for]
// questo lo fai per tutto quanto che ti serve.
// se tutto è apposto, solo allora vai a farti una funzione che carichi 
// la direzione che ti serva per sapere dove sparare il proiettile
```